### PR TITLE
drm-display: remove root privilege in run-time

### PR DESCRIPTION
**Pull request by liuyinhangx**
* drm-display: remove root privilege in run-time
 * smart-lib: do not compile sample-smart-lib if disable aiq

### CODE REVIEW

**Comment by liuyinhangx** [comment]
Update patch, please review.

File: tests/test-device-manager.cpp [hunk 1]
```diff
@@ -75,10 +75,11 @@ class MainDeviceManager
         , _frame_height (0)
         , _frame_count (0)
         , _frame_save (0)
-        , _enable_display (false)
+        , _need_display (false)
     {
 #if HAVE_LIBDRM
-        _display = DrmDisplay::instance();
+        _display = DrmDisplay::instance ();
+        _need_display = _display->get_preview ();
```
Comment (windyuan) [inline tests/test-device-manager.cpp]: how about keep _enable_display and add a new member _can_display. 
e.g.

    _can_display = _display->can_preview ();

File: tests/test-device-manager.cpp [hunk 2]
```diff
@@ -150,7 +147,7 @@ MainDeviceManager::handle_buffer (const SmartPtr<VideoBuffer> &buf)
 
     XCAM_OBJ_PROFILING_START;
 
-    if (_enable_display)
+    if (_need_display)
```
Comment (windyuan) [inline tests/test-device-manager.cpp]: if (_enable_display && _can_display)
     ....

File: tests/test-pipe-manager.cpp [hunk 3]
```diff
@@ -98,7 +95,7 @@ MainPipeManager::post_buffer (const SmartPtr<VideoBuffer> &buf)
 
     XCAM_OBJ_PROFILING_START;
 
-    if (_enable_display)
+    if (_need_display)
```
Comment (windyuan) [inline tests/test-pipe-manager.cpp]: same as TestDeviceManager

File: xcore/drm_display.h [hunk 4]
```diff
@@ -101,17 +102,25 @@ class DrmDisplay {
         const enum v4l2_buf_type buf_type);
     SmartPtr<DrmBoBuffer> convert_to_drm_bo_buf (SmartPtr<DrmDisplay> &self, SmartPtr<VideoBuffer> &buf_in);
 
-    bool set_display_mode(DrmDisplayMode mode) {
+    static bool set_preview (bool flag);
+    bool get_preview () {
```
Comment (windyuan) [inline xcore/drm_display.h]: can_preview

**Comment by windyuan** [comment]
merged. thanks.

commit 56cabfda1e6fcf8efe5ffa7aec49cff6d7c36393
commit 5855e6f0067b7b760670117ac7d602c655d942c7

**Pull request closed without merging**